packages/flutter/lib/src/material/button_style_button.dart

```diff
@@ -0,0 +1,470 @@
+    if (_disabled && _pressed) {
+      _handleHighlightChanged(false);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ButtonStyle widgetStyle = widget.style;
+    final ButtonStyle themeStyle = widget.themeStyleOf(context);
+    final ButtonStyle defaultStyle = widget.defaultStyleOf(context);
+    assert(defaultStyle != null);
+
+    T effectiveValue<T>(T Function(ButtonStyle style) getProperty) {
+      final T widgetValue  = getProperty(widgetStyle);
+      final T themeValue   = getProperty(themeStyle);
+      final T defaultValue = getProperty(defaultStyle);
+      return widgetValue ?? themeValue ?? defaultValue;
+    }
+
+    T resolve<T>(MaterialStateProperty<T> Function(ButtonStyle style) getProperty) {
+      return effectiveValue(
+        (ButtonStyle style) => getProperty(style)?.resolve(_states),
+      );
+    }
+
+    final TextStyle resolvedTextStyle = resolve<TextStyle>((ButtonStyle style) => style?.textStyle);
+    final Color resolvedBackgroundColor = resolve<Color>((ButtonStyle style) => style?.backgroundColor);
+    final Color resolvedForegroundColor = resolve<Color>((ButtonStyle style) => style?.foregroundColor);
+    final Color resolvedShadowColor = resolve<Color>((ButtonStyle style) => style?.shadowColor);
+    final double resolvedElevation = resolve<double>((ButtonStyle style) => style?.elevation);
+    final EdgeInsetsGeometry resolvedPadding = resolve<EdgeInsetsGeometry>((ButtonStyle style) => style?.padding);
+    final Size resolvedMinimumSize = resolve<Size>((ButtonStyle style) => style?.minimumSize);
+    final BorderSide resolvedSide = resolve<BorderSide>((ButtonStyle style) => style?.side);
+    final OutlinedBorder resolvedShape = resolve<OutlinedBorder>((ButtonStyle style) => style?.shape);
+
+    final MaterialStateMouseCursor resolvedMouseCursor = _MouseCursor(
+      (Set<MaterialState> states) => effectiveValue((ButtonStyle style) => style?.mouseCursor?.resolve(states)),
+    );
+
+    final MaterialStateProperty<Color> overlayColor = MaterialStateProperty.resolveWith<Color>(
+      (Set<MaterialState> states) => effectiveValue((ButtonStyle style) => style?.overlayColor?.resolve(states)),
+    );
+
+    final VisualDensity resolvedVisualDensity = effectiveValue((ButtonStyle style) => style?.visualDensity);
+    final MaterialTapTargetSize resolvedTapTargetSize = effectiveValue((ButtonStyle style) => style?.tapTargetSize);
+    final Duration resolvedAnimationDuration = effectiveValue((ButtonStyle style) => style?.animationDuration);
+    final bool resolvedEnableFeedback = effectiveValue((ButtonStyle style) => style?.enableFeedback);
+    final Offset densityAdjustment = resolvedVisualDensity.baseSizeAdjustment;
+    final BoxConstraints effectiveConstraints = resolvedVisualDensity.effectiveConstraints(
+      BoxConstraints(
+        minWidth: resolvedMinimumSize.width,
+        minHeight: resolvedMinimumSize.height,
+      ),
+    );
```

Comment:
@HansMuller What about let the user provide the minimum width/height as in the old buttons. I know we can always wrap it in a SizedBox but I think it would be great to have that option.

Reply:
Also, this can avoid one more wiget in the tree :)

Reply:
Specifying a minimum size is still supported: the `resolvedMinimumSize` you're referring to is based on the button style's minimumSize: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/button_style.dart#L160

Reply:
I meant the Widgets that depend on this one does not have a way to provide a custom size like the `ContainedButton` and `OutlinedButton` they does not have a `height` or minWidth property for example, just like the old buttons.

Reply:
It's true that the new buttons don't have explicit width or height parameters. However one can specify the minimum size for an individual button:

```dart
TextButton(
  style: TextButton.styleFrom(
    minimumSize: Size(100, 100),
  ),
  child: Text('I am at least 100x100')
)
```
For a fixed size button, you're right that it would be necessary to use a SizedBox:
```
TextButton(
  child: SizedBox(
   width: 100,
   height: 100,
   child: Text('I am 100x100'),
  ),
)
```

Reply:
Size is now parte of the button style?
Is the last piece of code valid? I mean you set the size for the child instead of the whole button widget.

I need to try this :)

Reply:
I think we can take advantage of the possibility of passing the widget size on its parent widget to allow users to customize the widget without the need for another widget.

Reply:
The `minimumSize` is part of ButtonStyle (it was part of the ButtonTheme too, see https://api.flutter.dev/flutter/material/ButtonThemeData/constraints.html).

The second example produces a fixed size button, although it's not exactly 100x100 because padding.

Reply:
Sounds right. Thank you.